substrate-archive-common/src/util.rs

```diff
@@ -138,3 +104,37 @@ fn format_opt(file: Option<String>) -> String {
         Some(f) => f,
     }
 }
+
+/// get the path to a local substrate directory where we can save data
+pub fn substrate_dir() -> PathBuf {
+    if let Some(base_dirs) = dirs::BaseDirs::new() {
+        let mut path = base_dirs.data_local_dir().to_path_buf();
+        path.push("substrate_archive");
+        path
+    } else {
+        panic!("Couldn't establish substrate data local path");
```

Comment:
Why panic here but `std::process::exit` in `create_dir`? Maybe the decision on what to do should be in the caller, so that we have a single place where decisions like exit/panic/retry/call mum can be made?

Reply:
The "error message" could also be improved, https://docs.rs/directories/3.0.1/directories/struct.BaseDirs.html#method.new fails if the home directory couldn't be found.